src/System.Collections.Concurrent/src/System/Collections/Concurrent/PartitionerStatic.cs

```diff
@@ -1160,13 +1160,13 @@ override protected bool HasNoElementsLeft
             {
                 get
                 {
-                    Contract.Assert(_sharedIndex != null);
+                    Debug.Assert(_sharedIndex != null);
                     // use the new Volatile.Read method because it is cheaper than Interlocked.Read on AMD64 architecture
                     return Volatile.Read(ref _sharedIndex.Value) >= SourceCount - 1;
                 }
                 set
                 {
-                    Contract.Assert(false);
+                    Debug.Assert(false);
```

Comment:
There are ~30 cases of Debug.Assert(false, and I believe all but this one (at least in the files you modified) already have a message argument.  Could we change all of them to use Debug.Fail instead?